### ISSUE TITLE
ArgoCD 1.7.6 updating allowed namespaces for a cluster from the UI broke the cluster config

### ISSUE THREAD

**Issue by rayanebel** [argoproj/argo-cd#4535]
**Describe the bug**

Hi,

I'am trying to deploy a new instance of argocd _(1.7.6)_. I know that when argocd is installed, it create a new cluster object for it's own kubernetes cluster _(in-cluster)_ and by default this cluster use the service account of `application-controller`. Now, I would like to overwrite it and replace the default service account to use another one.

@alexmt already explain me in the past and for a previous version that, we can override the default cluster by creating our own kubernetes secret with the URL: https://kubernetes.default.svc.

So I've created this secret with the good bearer token corresponding to another SA which are in another namespace (I've also created a rolebinding in the argocd namespace to allow the SA in remote namespace to access in the other namespaces).

In the UI, I see my cluster in Successfull state but now, if I try to edit the cluster from the UI to add another namespace in the whitelist, my cluster is now in Failure state and when I checked the details, I have this error:

```
failed to load initial state of resource ReplicaSet.apps: replicasets.apps is forbidden: User "system:serviceaccount:demo-argocd:argocd-application-controller" cannot list resource "replicasets" in API group "apps" in the namespace "demo-argocd"
```

I don't know why argocd is trying to use the service of `application-controller` (after the update on the ui) when I have configured this cluster to use another service account ?

When I rollback the configuration to one namespace in the whitelist _(by applying the secret with kubectl)_ for this cluster and I refresh the cache, my cluster back to a successful state.

**INFORMATIONS**

When I add more than one namespace for the cluster in the UI, it's not working but, If I add this additional namespaces directly in the kubernetes secret it's working.

Can you help me to understand what's going on ?

**To Reproduce**

- Override the default cluster to use another SA in another namespace
- Update this cluster from the UI and add more than one namespace in the whitelist
- Check cluster details

**Expected behavior**

A cluster in successful state.

**Version**

```shell
argocd-server: 1.7.6
```

**Logs**

```
argo-cd-argocd-application-controller-67c7859d7f-llqs8 application-controller E1012 13:33:11.768714       1 retrywatcher.go:129] Watch failed: unknown
argo-cd-argocd-application-controller-67c7859d7f-llqs8 application-controller E1012 13:33:11.768837       1 retrywatcher.go:129] Watch failed: unknown
argo-cd-argocd-application-controller-67c7859d7f-llqs8 application-controller time="2020-10-12T13:33:11Z" level=error msg="Failed to sync cluster https://10.0.0.1:443: failed to load initial state of resource PersistentVolumeClaim: persistentvolumeclaims is forbidden: User \"system:serviceaccount:demo-argocd:argocd-application-controller\" cannot list resource \"persistentvolumeclaims\" in API group \"\" in the namespace \"demo-argocd\""
argo-cd-argocd-application-controller-67c7859d7f-llqs8 application-controller E1012 13:33:12.116041       1 retrywatcher.go:129] Watch failed: unknown
```

**Comment by jessesuen** [comment]
This is a duplicate to something that will be fixed in v1.8 (alex will find the dup)

**Comment by alexmt** [comment]
Closing as duplicate of #4405 (fix is on the way: https://github.com/argoproj/argo-cd/pull/4553)
